docs/proposals/Release-Supply-Chain-Metadata.md

```diff
@@ -0,0 +1,174 @@
+# Supply Chain Metadata for Ratify Assets
+
+Ratify currently publishes multiple forms of release assets both for production and development uses. Currently, these assets are not published with accompanying supply chain metadata including signatures, SBOMs, and provenance. Shipping each of these forms of metadata with all binaries and container images produced by Ratify will provide consumers a verifiable way to guarantee integrity of Ratify assets. Furthermore, this will improve Ratify's OSSF scorecard.
+
+## What does Ratify currently publish?
+
+Ratify publishes two types: release and development. Release assets accompany official Ratify Github releases. Development assets are published weekly (or adhoc as needed).
+
+Each publish type includes the following group of assets:
+
+* CRD container image to `ghcr.io/ratify-project`
+* Base container image to `ghcr.io/ratify-project`
+* Base + plugins container image to `ghcr.io/ratify-project`
+* Ratify binaries as a single bundle per OS/arch which includes:
+  * ratify binary
+  * sbom-verifier plugin binary
+  * vulernability-report verifier plugin binary
+* `checksums.txt` which contains list of checksum for each binary bundle
+* Helm chart
+  * Release ONLY: published packaged helm chart to `ratify-project.github.io`
+  * Dev ONLY: published packaged helm chart to `ghcr.io/ratify-project`
+* Source code (zip + tar ball)
+
+## Signatures
+
+Ratify should support signing all container images and binaries with both Sigstore Cosign and Notary Project.
+
+### Notary Project Image Signatures
+
+Ratify can utilize Notary Project's install and sign github actions published.
+Notation requires a KMS to store the signing certificate. There is no support for Github Encrypted Secrets for storing the certificate.
+
+1. Generate a self-signed certificate stored in Azure Key Vault
+2. Configure Notation action to sign using the cert stored in AKV
+3. Sign CRD, base, plugin images
+4. Store the public verification certificate at `ratify.dev/.well-known/ratify-verification.crt` & `github.com/ratify-project/ratify/dev/.well-known/ratify-verification.crt`
```

Comment:
it's currently a 1 year expiry. It will auto renew in AKV however I think that means we need to upload a new pubic certificate right?

Reply:
yeah, that's true. Also I'm wondering if we donate to CNCF, do we need to move from AKV? @FeynmanZhou do you have insights on it?

Reply:
Along those same lines, @FeynmanZhou how do we distribute the different versioned certs. We can publish latest cert which user can use to verify latest binary but how are they supposed to verify previous versions that use different public cert?

Reply:
This concern is valid. We can always use the `latest` version of the AKV key and cert to sign and verify. Notation Azure Key Vault plugin recently supports retrieving the latest version of a key or cert without specifying version number. Relevant issue https://github.com/Azure/notation-azure-kv/issues/180. 

If users pass the URI `https://<vaultname>.vault.azure.net/<key|certificate>/<name>` to Notation AKV plugin, it should point to the latest version of the key/cert. 

We need to figure out how to distribute the latest cert to users so that they can verify easier.

Reply:
The `.well-known` endpoint is the easiest way to distribute certs. You can also name the cert file with versions.
For example,
```
ratify.dev/.well-known/ratify-verification.crt
ratify.dev/.well-known/ratify-verification_2024.crt
ratify.dev/.well-known/ratify-verification_2025.crt
```

Reply:
Good idea @shizhMSFT. I'm thinking for the versioning, maybe we set this convention:

The latest certificate will be stored at `ratify.dev/.well-known/pki-validation/ratify-verification.crt`. When a new version of that certificate is created & the new version is now used to sign all assets moving forward, we will update the `ratify.dev/.well-known/pki-validation/ratify-verification.crt` contents to be the new version. We will also add the old version as a new .crt file as `ratify.dev/.well-known/pki-validation/ratify-verification_<YYYYMMDD>.crt` where the timestamp is the date it stopped being used as latest.

What do you think?

Reply:
LGTM